src/Simulation/CsharpGeneration.Tests/SimulationCodeTests.fs

```diff
@@ -3204,3 +3203,112 @@ namespace Microsoft.Quantum.Tests.LineNumbers
+[assembly: SpecializationDeclaration("{\"Kind\":{\"Case\":\"QsBody\"},\"TypeArguments\":{\"Case\":\"Null\"},\"Information\":{\"Characteristics\":{\"Case\":\"EmptySet\"},\"InferredInformation\":{\"IsSelfAdjoint\":false,\"IsIntrinsic\":false}},\"Parent\":{\"Namespace\":\"Microsoft.Quantum.Tests.UnitTests\",\"Name\":\"UnitTest1\"},\"Attributes\":[],\"SourceFile\":\"%%%\",\"Position\":{\"Item1\":20,\"Item2\":4},\"HeaderRange\":{\"Item1\":{\"Line\":1,\"Column\":11},\"Item2\":{\"Line\":1,\"Column\":20}},\"Documentation\":[]}")]
+
+#line hidden
+namespace Microsoft.Quantum.Core
+{
+    public class Attribute : UDTBase<QVoid>, IApplyData
+    {
+        public Attribute() : base(default(QVoid))
+        {
+        }
+
+        public Attribute(QVoid data) : base(data)
+        {
+        }
+
+        System.Collections.Generic.IEnumerable<Qubit> IApplyData.Qubits => null;
+        public void Deconstruct()
+        {
+        }
+    }
+}
+
+#line hidden
+namespace Microsoft.Quantum.Diagnostics
+{
+    public class TestOperation : UDTBase<String>, IApplyData
+    {
+        public TestOperation() : base(default(String))
+        {
+        }
+
+        public TestOperation(String data) : base(data)
+        {
+        }
+
+        System.Collections.Generic.IEnumerable<Qubit> IApplyData.Qubits => null;
+        public void Deconstruct()
+        {
+        }
+    }
+}
+
+#line hidden
+namespace Microsoft.Quantum.Tests.UnitTests
+{
+    [SourceLocation("%%%", OperationFunctor.Body, 21, -1)]
+    public partial class UnitTest1 : Operation<QVoid, QVoid>, ICallable
+    {
+        public UnitTest1(IOperationFactory m) : base(m)
+        {
+        }
+
+        String ICallable.Name => "UnitTest1";
+        String ICallable.FullName => "Microsoft.Quantum.Tests.UnitTests.UnitTest1";
+        public override Func<QVoid, QVoid> Body => (__in__) =>
+        {
+#line hidden
+            return QVoid.Instance;
+        }
+
+        ;
+        public override void Init()
+        {
+        }
+
+        public override IApplyData __dataIn(QVoid data) => data;
+        public override IApplyData __dataOut(QVoid data) => data;
+        public static System.Threading.Tasks.Task<QVoid> Run(IOperationFactory __m__)
+        {
+            return __m__.Run<UnitTest1, QVoid, QVoid>(QVoid.Instance);
+        }
+    }
+
+    public partial class QuantumSimulatorTests
+    {
+        [Fact(DisplayName = "UnitTest1")]
+        public void __UnitTest1Test__()
+#line 20 "%%%"
+        {
+            var sim = new QuantumSimulator();
```

Comment:
I would provide a mechanism such that the simulator to use can be programmatically specified.
Maybe you could pass it during constructor? Or maybe creating and initializing is a protected method that can be overriden?

Reply:
When decorate a Q# operation with the TestsOperation attribute, you will be expected to provide the name of the target as an argument to the attribute. This name will appear here.